openmdao/jacobians/dictionary_jacobian.py

```diff
@@ -80,64 +80,79 @@ def _apply(self, d_inputs, d_outputs, d_residuals, mode):
         mode : str
             'fwd' or 'rev'.
         """
+        # avoid circular import
+        from openmdao.core.explicitcomponent import ExplicitComponent
```

Comment:
I'm not sure how expensive this is, but in case it's not negligible, what about importing ExplicitComponent once in __init__ and saving it as an attribute?